machineconfiguration/v1/types.go

```diff
@@ -421,6 +421,38 @@ type MachineConfigPoolSpec struct {
 	// The targeted MachineConfig object for the machine config pool.
 	// +optional
 	Configuration MachineConfigPoolStatusConfiguration `json:"configuration"`
+
+	// pinnedImageSets specifies a sequence of PinnedImageSetRef objects for the
+	// pool. Nodes within this pool will preload and pin images defined in the
+	// PinnedImageSet. Before pulling images the controller will ensure the
+	// total uncompressed size of all the images does not exceed available
```

Comment:
What happens if we do exceed the available disk space?

Reply:
controller does a precheck calculating the storage required for all images + buffer ~20GB - allocatable, before we pull. If this fails the pool goes degraded. The node is checked for KubeletHasNoDiskPressure if that at anytime is `true` controller workers will bail and retry if that can't be reconciled eventually it will be Unreconcilable. Also the node will become NotReady as a result of disk pressure (pretty sure).

Kubelet would force a GC on images(need to verify) and potentially evict pods. On the control-plane worst case it could take the cluster down. So there is a fair amount of assumption that the kubelet reports allocatable storage correctly.

Reply:
For this reason, we will not pin the images until all images are pulled to allow the Kubelet the ability to GC>